nixos/modules/services/matrix/draupnir.nix

```diff
@@ -0,0 +1,279 @@
+                  readOnly = true;
+                  description = ''
+                    Account name on the Matrix homeserver. Configured in `services.draupnir.pantalaimon.username`.
+                  '';
+                };
+              };
+            };
+          };
+          #endregion
+
+          #region Base settings
+          homeserverUrl = lib.mkOption {
+            type = lib.types.str;
+            defaultText = ''
+              URI to the pantalaimon instance, if configured. Else unset.
+            '';
+            description = ''
+              Base URL of the Matrix homeserver, that provides the Client-Server API.
+              If `services.draupnir.pantalaimon.enable` is `true`, this option will become read only. Configure `services.draupnir.pantalaimon.options.homeserver` instead in that case.
+              The listen address of `pantalaimon` will then become the `homeserverUrl` of `draupnir`.
+            '';
+          } // lib.optionalAttrs cfg.pantalaimon.enable {
+            default = "http://${config.services.pantalaimon-headless.instances."draupnir".listenAddress}:${toString config.services.pantalaimon-headless.instances."draupnir".listenPort}/";
+          };
+          #endregion
+
+          #region Common settings
+          autojoinOnlyIfManager = lib.mkOption {
+            type = lib.types.bool;
+            default = false;
+            description = ''
+              If `true`, the bot will only autojoin rooms if the user is a manager.
+            '';
+          };
+
+          automaticallyRedactForReasons = lib.mkOption {
+            type = lib.types.listOf lib.types.str;
+            default = [ ];
+            description = ''
+              A list of reasons for which the bot will automatically redact messages.
+            '';
+          };
+          managementRoom = lib.mkOption {
+            type = lib.types.str;
+            default = "#moderators:example.org";
+            description = ''
+              The room ID or alias where moderators can use the bot's functionality.
+
+              The bot has no access controls, so anyone in this room can use the bot - secure this room!
+
+              Warning: When using a room alias, make sure the alias used is on the local homeserver!
+              This prevents an issue where the control room becomes undefined when the alias can't be resolved.
+            '';
+          };
+
+          protectedRooms = lib.mkOption {
+            type = lib.types.listOf lib.types.str;
+            default = [ ];
+            example = lib.literalExpression ''
+              [
+                "https://matrix.to/#/#yourroom:example.org"
+                "https://matrix.to/#/#anotherroom:example.org"
+              ]
+            '';
+            description = ''
+              A list of rooms to protect (matrix.to URLs).
+              These can also be configured interactively.
+
+              Note that this option does nothing in Draupnir v2+!
```

Comment:
This notice is obsolete, since we're starting with 2.0+

Reply:
I wonder if we should keep it for users who choose to override the package with 1.x? Or should we remove the option completely?

Reply:
I don't think the module needs to account for all kinds of overrides people will be doing. I'd say keep the option (to set a package) and drop the notice.

Reply:
And thus drop the protectedRooms option being defined explicitly since it does nothing?

Reply:
Oh yeah, of course!